// xero-node/dist/gen/model/payroll-uk/statutoryDeductionCategory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StatutoryDeductionCategory = void 0;
/**
* Statutory Deduction Category
*/
var StatutoryDeductionCategory;
(function (StatutoryDeductionCategory) {
    StatutoryDeductionCategory[StatutoryDeductionCategory["AdditionalStudentLoan"] = 'AdditionalStudentLoan'] = "AdditionalStudentLoan";
    StatutoryDeductionCategory[StatutoryDeductionCategory["ChildSupport"] = 'ChildSupport'] = "ChildSupport";
    StatutoryDeductionCategory[StatutoryDeductionCategory["CourtFines"] = 'CourtFines'] = "CourtFines";
    StatutoryDeductionCategory[StatutoryDeductionCategory["Creditor"] = 'Creditor'] = "Creditor";
    StatutoryDeductionCategory[StatutoryDeductionCategory["FederalLevy"] = 'FederalLevy'] = "FederalLevy";
    StatutoryDeductionCategory[StatutoryDeductionCategory["InlandRevenueArrears"] = 'InlandRevenueArrears'] = "InlandRevenueArrears";
    StatutoryDeductionCategory[StatutoryDeductionCategory["KiwiSaver"] = 'KiwiSaver'] = "KiwiSaver";
    StatutoryDeductionCategory[StatutoryDeductionCategory["MsdRepayments"] = 'MsdRepayments'] = "MsdRepayments";
    StatutoryDeductionCategory[StatutoryDeductionCategory["NonPriorityOrder"] = 'NonPriorityOrder'] = "NonPriorityOrder";
    StatutoryDeductionCategory[StatutoryDeductionCategory["PriorityOrder"] = 'PriorityOrder'] = "PriorityOrder";
    StatutoryDeductionCategory[StatutoryDeductionCategory["TableBased"] = 'TableBased'] = "TableBased";
    StatutoryDeductionCategory[StatutoryDeductionCategory["StudentLoan"] = 'StudentLoan'] = "StudentLoan";
    StatutoryDeductionCategory[StatutoryDeductionCategory["VoluntaryStudentLoan"] = 'VoluntaryStudentLoan'] = "VoluntaryStudentLoan";
    StatutoryDeductionCategory[StatutoryDeductionCategory["USChildSupport"] = 'USChildSupport'] = "USChildSupport";
})(StatutoryDeductionCategory = exports.StatutoryDeductionCategory || (exports.StatutoryDeductionCategory = {}));
