// xero-node/dist/gen/model/payroll-nz/taxCode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxCode = void 0;
/**
* Tax codes used for employee tax
*/
var TaxCode;
(function (TaxCode) {
    TaxCode[TaxCode["ND"] = 'ND'] = "ND";
    TaxCode[TaxCode["M"] = 'M'] = "M";
    TaxCode[TaxCode["ME"] = 'ME'] = "ME";
    TaxCode[TaxCode["MSL"] = 'MSL'] = "MSL";
    TaxCode[TaxCode["MESL"] = 'MESL'] = "MESL";
    TaxCode[TaxCode["SB"] = 'SB'] = "SB";
    TaxCode[TaxCode["S"] = 'S'] = "S";
    TaxCode[TaxCode["SH"] = 'SH'] = "SH";
    TaxCode[TaxCode["ST"] = 'ST'] = "ST";
    TaxCode[TaxCode["SBSL"] = 'SBSL'] = "SBSL";
    TaxCode[TaxCode["SSL"] = 'SSL'] = "SSL";
    TaxCode[TaxCode["SHSL"] = 'SHSL'] = "SHSL";
    TaxCode[TaxCode["STSL"] = 'STSL'] = "STSL";
    TaxCode[TaxCode["WT"] = 'WT'] = "WT";
    TaxCode[TaxCode["CAE"] = 'CAE'] = "CAE";
    TaxCode[TaxCode["EDW"] = 'EDW'] = "EDW";
    TaxCode[TaxCode["NSW"] = 'NSW'] = "NSW";
    TaxCode[TaxCode["STC"] = 'STC'] = "STC";
    TaxCode[TaxCode["STCSL"] = 'STCSL'] = "STCSL";
})(TaxCode = exports.TaxCode || (exports.TaxCode = {}));
