// xero-node/dist/gen/model/payroll-au/countryOfResidence.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountryOfResidence = void 0;
/**
* Country of residence as a valid ISO 3166-1 alpha-2 country code e.g. \"AU\", \"NZ\", \"CA\" or an empty string (\"\") to unset a previously set value. Only applicable, and mandatory if income type is WORKINGHOLIDAYMAKER.
*/
var CountryOfResidence;
(function (CountryOfResidence) {
    CountryOfResidence[CountryOfResidence["AF"] = 'AF'] = "AF";
    CountryOfResidence[CountryOfResidence["AX"] = 'AX'] = "AX";
    CountryOfResidence[CountryOfResidence["AL"] = 'AL'] = "AL";
    CountryOfResidence[CountryOfResidence["DZ"] = 'DZ'] = "DZ";
    CountryOfResidence[CountryOfResidence["AS"] = 'AS'] = "AS";
    CountryOfResidence[CountryOfResidence["AD"] = 'AD'] = "AD";
    CountryOfResidence[CountryOfResidence["AO"] = 'AO'] = "AO";
    CountryOfResidence[CountryOfResidence["AI"] = 'AI'] = "AI";
    CountryOfResidence[CountryOfResidence["AQ"] = 'AQ'] = "AQ";
    CountryOfResidence[CountryOfResidence["AG"] = 'AG'] = "AG";
    CountryOfResidence[CountryOfResidence["AR"] = 'AR'] = "AR";
    CountryOfResidence[CountryOfResidence["AM"] = 'AM'] = "AM";
    CountryOfResidence[CountryOfResidence["AW"] = 'AW'] = "AW";
    CountryOfResidence[CountryOfResidence["AU"] = 'AU'] = "AU";
    CountryOfResidence[CountryOfResidence["AT"] = 'AT'] = "AT";
    CountryOfResidence[CountryOfResidence["AZ"] = 'AZ'] = "AZ";
    CountryOfResidence[CountryOfResidence["BS"] = 'BS'] = "BS";
    CountryOfResidence[CountryOfResidence["BH"] = 'BH'] = "BH";
    CountryOfResidence[CountryOfResidence["BD"] = 'BD'] = "BD";
    CountryOfResidence[CountryOfResidence["BB"] = 'BB'] = "BB";
    CountryOfResidence[CountryOfResidence["BY"] = 'BY'] = "BY";
    CountryOfResidence[CountryOfResidence["BE"] = 'BE'] = "BE";
    CountryOfResidence[CountryOfResidence["BZ"] = 'BZ'] = "BZ";
    CountryOfResidence[CountryOfResidence["BJ"] = 'BJ'] = "BJ";
    CountryOfResidence[CountryOfResidence["BM"] = 'BM'] = "BM";
    CountryOfResidence[CountryOfResidence["BT"] = 'BT'] = "BT";
    CountryOfResidence[CountryOfResidence["BO"] = 'BO'] = "BO";
    CountryOfResidence[CountryOfResidence["BA"] = 'BA'] = "BA";
    CountryOfResidence[CountryOfResidence["BW"] = 'BW'] = "BW";
    CountryOfResidence[CountryOfResidence["BV"] = 'BV'] = "BV";
    CountryOfResidence[CountryOfResidence["BR"] = 'BR'] = "BR";
    CountryOfResidence[CountryOfResidence["IO"] = 'IO'] = "IO";
    CountryOfResidence[CountryOfResidence["BN"] = 'BN'] = "BN";
    CountryOfResidence[CountryOfResidence["BG"] = 'BG'] = "BG";
    CountryOfResidence[CountryOfResidence["BF"] = 'BF'] = "BF";
    CountryOfResidence[CountryOfResidence["BI"] = 'BI'] = "BI";
    CountryOfResidence[CountryOfResidence["KH"] = 'KH'] = "KH";
    CountryOfResidence[CountryOfResidence["CM"] = 'CM'] = "CM";
    CountryOfResidence[CountryOfResidence["CA"] = 'CA'] = "CA";
    CountryOfResidence[CountryOfResidence["CV"] = 'CV'] = "CV";
    CountryOfResidence[CountryOfResidence["KY"] = 'KY'] = "KY";
    CountryOfResidence[CountryOfResidence["CF"] = 'CF'] = "CF";
    CountryOfResidence[CountryOfResidence["TD"] = 'TD'] = "TD";
    CountryOfResidence[CountryOfResidence["CL"] = 'CL'] = "CL";
    CountryOfResidence[CountryOfResidence["CN"] = 'CN'] = "CN";
    CountryOfResidence[CountryOfResidence["CX"] = 'CX'] = "CX";
    CountryOfResidence[CountryOfResidence["CC"] = 'CC'] = "CC";
    CountryOfResidence[CountryOfResidence["CO"] = 'CO'] = "CO";
    CountryOfResidence[CountryOfResidence["KM"] = 'KM'] = "KM";
    CountryOfResidence[CountryOfResidence["CG"] = 'CG'] = "CG";
    CountryOfResidence[CountryOfResidence["CD"] = 'CD'] = "CD";
    CountryOfResidence[CountryOfResidence["CK"] = 'CK'] = "CK";
    CountryOfResidence[CountryOfResidence["CR"] = 'CR'] = "CR";
    CountryOfResidence[CountryOfResidence["CI"] = 'CI'] = "CI";
    CountryOfResidence[CountryOfResidence["HR"] = 'HR'] = "HR";
    CountryOfResidence[CountryOfResidence["CU"] = 'CU'] = "CU";
    CountryOfResidence[CountryOfResidence["CY"] = 'CY'] = "CY";
    CountryOfResidence[CountryOfResidence["CZ"] = 'CZ'] = "CZ";
    CountryOfResidence[CountryOfResidence["DK"] = 'DK'] = "DK";
    CountryOfResidence[CountryOfResidence["DJ"] = 'DJ'] = "DJ";
    CountryOfResidence[CountryOfResidence["DM"] = 'DM'] = "DM";
    CountryOfResidence[CountryOfResidence["DO"] = 'DO'] = "DO";
    CountryOfResidence[CountryOfResidence["EC"] = 'EC'] = "EC";
    CountryOfResidence[CountryOfResidence["EG"] = 'EG'] = "EG";
    CountryOfResidence[CountryOfResidence["SV"] = 'SV'] = "SV";
    CountryOfResidence[CountryOfResidence["GQ"] = 'GQ'] = "GQ";
    CountryOfResidence[CountryOfResidence["ER"] = 'ER'] = "ER";
    CountryOfResidence[CountryOfResidence["EE"] = 'EE'] = "EE";
    CountryOfResidence[CountryOfResidence["ET"] = 'ET'] = "ET";
    CountryOfResidence[CountryOfResidence["FK"] = 'FK'] = "FK";
    CountryOfResidence[CountryOfResidence["FO"] = 'FO'] = "FO";
    CountryOfResidence[CountryOfResidence["FJ"] = 'FJ'] = "FJ";
    CountryOfResidence[CountryOfResidence["FI"] = 'FI'] = "FI";
    CountryOfResidence[CountryOfResidence["FR"] = 'FR'] = "FR";
    CountryOfResidence[CountryOfResidence["GF"] = 'GF'] = "GF";
    CountryOfResidence[CountryOfResidence["PF"] = 'PF'] = "PF";
    CountryOfResidence[CountryOfResidence["TF"] = 'TF'] = "TF";
    CountryOfResidence[CountryOfResidence["GA"] = 'GA'] = "GA";
    CountryOfResidence[CountryOfResidence["GM"] = 'GM'] = "GM";
    CountryOfResidence[CountryOfResidence["GE"] = 'GE'] = "GE";
    CountryOfResidence[CountryOfResidence["DE"] = 'DE'] = "DE";
    CountryOfResidence[CountryOfResidence["GH"] = 'GH'] = "GH";
    CountryOfResidence[CountryOfResidence["GI"] = 'GI'] = "GI";
    CountryOfResidence[CountryOfResidence["GR"] = 'GR'] = "GR";
    CountryOfResidence[CountryOfResidence["GL"] = 'GL'] = "GL";
    CountryOfResidence[CountryOfResidence["GD"] = 'GD'] = "GD";
    CountryOfResidence[CountryOfResidence["GP"] = 'GP'] = "GP";
    CountryOfResidence[CountryOfResidence["GU"] = 'GU'] = "GU";
    CountryOfResidence[CountryOfResidence["GT"] = 'GT'] = "GT";
    CountryOfResidence[CountryOfResidence["GG"] = 'GG'] = "GG";
    CountryOfResidence[CountryOfResidence["GN"] = 'GN'] = "GN";
    CountryOfResidence[CountryOfResidence["GW"] = 'GW'] = "GW";
    CountryOfResidence[CountryOfResidence["GY"] = 'GY'] = "GY";
    CountryOfResidence[CountryOfResidence["HT"] = 'HT'] = "HT";
    CountryOfResidence[CountryOfResidence["HM"] = 'HM'] = "HM";
    CountryOfResidence[CountryOfResidence["VA"] = 'VA'] = "VA";
    CountryOfResidence[CountryOfResidence["HN"] = 'HN'] = "HN";
    CountryOfResidence[CountryOfResidence["HK"] = 'HK'] = "HK";
    CountryOfResidence[CountryOfResidence["HU"] = 'HU'] = "HU";
    CountryOfResidence[CountryOfResidence["IS"] = 'IS'] = "IS";
    CountryOfResidence[CountryOfResidence["IN"] = 'IN'] = "IN";
    CountryOfResidence[CountryOfResidence["ID"] = 'ID'] = "ID";
    CountryOfResidence[CountryOfResidence["IR"] = 'IR'] = "IR";
    CountryOfResidence[CountryOfResidence["IQ"] = 'IQ'] = "IQ";
    CountryOfResidence[CountryOfResidence["IE"] = 'IE'] = "IE";
    CountryOfResidence[CountryOfResidence["IM"] = 'IM'] = "IM";
    CountryOfResidence[CountryOfResidence["IL"] = 'IL'] = "IL";
    CountryOfResidence[CountryOfResidence["IT"] = 'IT'] = "IT";
    CountryOfResidence[CountryOfResidence["JM"] = 'JM'] = "JM";
    CountryOfResidence[CountryOfResidence["JP"] = 'JP'] = "JP";
    CountryOfResidence[CountryOfResidence["JE"] = 'JE'] = "JE";
    CountryOfResidence[CountryOfResidence["JO"] = 'JO'] = "JO";
    CountryOfResidence[CountryOfResidence["KZ"] = 'KZ'] = "KZ";
    CountryOfResidence[CountryOfResidence["KE"] = 'KE'] = "KE";
    CountryOfResidence[CountryOfResidence["KI"] = 'KI'] = "KI";
    CountryOfResidence[CountryOfResidence["KP"] = 'KP'] = "KP";
    CountryOfResidence[CountryOfResidence["KR"] = 'KR'] = "KR";
    CountryOfResidence[CountryOfResidence["KW"] = 'KW'] = "KW";
    CountryOfResidence[CountryOfResidence["KG"] = 'KG'] = "KG";
    CountryOfResidence[CountryOfResidence["LA"] = 'LA'] = "LA";
    CountryOfResidence[CountryOfResidence["LV"] = 'LV'] = "LV";
    CountryOfResidence[CountryOfResidence["LB"] = 'LB'] = "LB";
    CountryOfResidence[CountryOfResidence["LS"] = 'LS'] = "LS";
    CountryOfResidence[CountryOfResidence["LR"] = 'LR'] = "LR";
    CountryOfResidence[CountryOfResidence["LY"] = 'LY'] = "LY";
    CountryOfResidence[CountryOfResidence["LI"] = 'LI'] = "LI";
    CountryOfResidence[CountryOfResidence["LT"] = 'LT'] = "LT";
    CountryOfResidence[CountryOfResidence["LU"] = 'LU'] = "LU";
    CountryOfResidence[CountryOfResidence["MO"] = 'MO'] = "MO";
    CountryOfResidence[CountryOfResidence["MK"] = 'MK'] = "MK";
    CountryOfResidence[CountryOfResidence["MG"] = 'MG'] = "MG";
    CountryOfResidence[CountryOfResidence["MW"] = 'MW'] = "MW";
    CountryOfResidence[CountryOfResidence["MY"] = 'MY'] = "MY";
    CountryOfResidence[CountryOfResidence["MV"] = 'MV'] = "MV";
    CountryOfResidence[CountryOfResidence["ML"] = 'ML'] = "ML";
    CountryOfResidence[CountryOfResidence["MT"] = 'MT'] = "MT";
    CountryOfResidence[CountryOfResidence["MH"] = 'MH'] = "MH";
    CountryOfResidence[CountryOfResidence["MQ"] = 'MQ'] = "MQ";
    CountryOfResidence[CountryOfResidence["MR"] = 'MR'] = "MR";
    CountryOfResidence[CountryOfResidence["MU"] = 'MU'] = "MU";
    CountryOfResidence[CountryOfResidence["YT"] = 'YT'] = "YT";
    CountryOfResidence[CountryOfResidence["MX"] = 'MX'] = "MX";
    CountryOfResidence[CountryOfResidence["FM"] = 'FM'] = "FM";
    CountryOfResidence[CountryOfResidence["MD"] = 'MD'] = "MD";
    CountryOfResidence[CountryOfResidence["MC"] = 'MC'] = "MC";
    CountryOfResidence[CountryOfResidence["MN"] = 'MN'] = "MN";
    CountryOfResidence[CountryOfResidence["ME"] = 'ME'] = "ME";
    CountryOfResidence[CountryOfResidence["MS"] = 'MS'] = "MS";
    CountryOfResidence[CountryOfResidence["MA"] = 'MA'] = "MA";
    CountryOfResidence[CountryOfResidence["MZ"] = 'MZ'] = "MZ";
    CountryOfResidence[CountryOfResidence["MM"] = 'MM'] = "MM";
    CountryOfResidence[CountryOfResidence["NA"] = 'NA'] = "NA";
    CountryOfResidence[CountryOfResidence["NR"] = 'NR'] = "NR";
    CountryOfResidence[CountryOfResidence["NP"] = 'NP'] = "NP";
    CountryOfResidence[CountryOfResidence["NL"] = 'NL'] = "NL";
    CountryOfResidence[CountryOfResidence["AN"] = 'AN'] = "AN";
    CountryOfResidence[CountryOfResidence["NC"] = 'NC'] = "NC";
    CountryOfResidence[CountryOfResidence["NZ"] = 'NZ'] = "NZ";
    CountryOfResidence[CountryOfResidence["NI"] = 'NI'] = "NI";
    CountryOfResidence[CountryOfResidence["NE"] = 'NE'] = "NE";
    CountryOfResidence[CountryOfResidence["NG"] = 'NG'] = "NG";
    CountryOfResidence[CountryOfResidence["NU"] = 'NU'] = "NU";
    CountryOfResidence[CountryOfResidence["NF"] = 'NF'] = "NF";
    CountryOfResidence[CountryOfResidence["MP"] = 'MP'] = "MP";
    CountryOfResidence[CountryOfResidence["NO"] = 'NO'] = "NO";
    CountryOfResidence[CountryOfResidence["OM"] = 'OM'] = "OM";
    CountryOfResidence[CountryOfResidence["PK"] = 'PK'] = "PK";
    CountryOfResidence[CountryOfResidence["PW"] = 'PW'] = "PW";
    CountryOfResidence[CountryOfResidence["PS"] = 'PS'] = "PS";
    CountryOfResidence[CountryOfResidence["PA"] = 'PA'] = "PA";
    CountryOfResidence[CountryOfResidence["PG"] = 'PG'] = "PG";
    CountryOfResidence[CountryOfResidence["PY"] = 'PY'] = "PY";
    CountryOfResidence[CountryOfResidence["PE"] = 'PE'] = "PE";
    CountryOfResidence[CountryOfResidence["PH"] = 'PH'] = "PH";
    CountryOfResidence[CountryOfResidence["PN"] = 'PN'] = "PN";
    CountryOfResidence[CountryOfResidence["PL"] = 'PL'] = "PL";
    CountryOfResidence[CountryOfResidence["PT"] = 'PT'] = "PT";
    CountryOfResidence[CountryOfResidence["PR"] = 'PR'] = "PR";
    CountryOfResidence[CountryOfResidence["QA"] = 'QA'] = "QA";
    CountryOfResidence[CountryOfResidence["RE"] = 'RE'] = "RE";
    CountryOfResidence[CountryOfResidence["RO"] = 'RO'] = "RO";
    CountryOfResidence[CountryOfResidence["RU"] = 'RU'] = "RU";
    CountryOfResidence[CountryOfResidence["RW"] = 'RW'] = "RW";
    CountryOfResidence[CountryOfResidence["BL"] = 'BL'] = "BL";
    CountryOfResidence[CountryOfResidence["SH"] = 'SH'] = "SH";
    CountryOfResidence[CountryOfResidence["KN"] = 'KN'] = "KN";
    CountryOfResidence[CountryOfResidence["LC"] = 'LC'] = "LC";
    CountryOfResidence[CountryOfResidence["MF"] = 'MF'] = "MF";
    CountryOfResidence[CountryOfResidence["PM"] = 'PM'] = "PM";
    CountryOfResidence[CountryOfResidence["VC"] = 'VC'] = "VC";
    CountryOfResidence[CountryOfResidence["WS"] = 'WS'] = "WS";
    CountryOfResidence[CountryOfResidence["SM"] = 'SM'] = "SM";
    CountryOfResidence[CountryOfResidence["ST"] = 'ST'] = "ST";
    CountryOfResidence[CountryOfResidence["SA"] = 'SA'] = "SA";
    CountryOfResidence[CountryOfResidence["SN"] = 'SN'] = "SN";
    CountryOfResidence[CountryOfResidence["RS"] = 'RS'] = "RS";
    CountryOfResidence[CountryOfResidence["SC"] = 'SC'] = "SC";
    CountryOfResidence[CountryOfResidence["SL"] = 'SL'] = "SL";
    CountryOfResidence[CountryOfResidence["SG"] = 'SG'] = "SG";
    CountryOfResidence[CountryOfResidence["SK"] = 'SK'] = "SK";
    CountryOfResidence[CountryOfResidence["SI"] = 'SI'] = "SI";
    CountryOfResidence[CountryOfResidence["SB"] = 'SB'] = "SB";
    CountryOfResidence[CountryOfResidence["SO"] = 'SO'] = "SO";
    CountryOfResidence[CountryOfResidence["ZA"] = 'ZA'] = "ZA";
    CountryOfResidence[CountryOfResidence["GS"] = 'GS'] = "GS";
    CountryOfResidence[CountryOfResidence["ES"] = 'ES'] = "ES";
    CountryOfResidence[CountryOfResidence["LK"] = 'LK'] = "LK";
    CountryOfResidence[CountryOfResidence["SD"] = 'SD'] = "SD";
    CountryOfResidence[CountryOfResidence["SR"] = 'SR'] = "SR";
    CountryOfResidence[CountryOfResidence["SJ"] = 'SJ'] = "SJ";
    CountryOfResidence[CountryOfResidence["SZ"] = 'SZ'] = "SZ";
    CountryOfResidence[CountryOfResidence["SE"] = 'SE'] = "SE";
    CountryOfResidence[CountryOfResidence["CH"] = 'CH'] = "CH";
    CountryOfResidence[CountryOfResidence["SY"] = 'SY'] = "SY";
    CountryOfResidence[CountryOfResidence["TW"] = 'TW'] = "TW";
    CountryOfResidence[CountryOfResidence["TJ"] = 'TJ'] = "TJ";
    CountryOfResidence[CountryOfResidence["TZ"] = 'TZ'] = "TZ";
    CountryOfResidence[CountryOfResidence["TH"] = 'TH'] = "TH";
    CountryOfResidence[CountryOfResidence["TL"] = 'TL'] = "TL";
    CountryOfResidence[CountryOfResidence["TG"] = 'TG'] = "TG";
    CountryOfResidence[CountryOfResidence["TK"] = 'TK'] = "TK";
    CountryOfResidence[CountryOfResidence["TO"] = 'TO'] = "TO";
    CountryOfResidence[CountryOfResidence["TT"] = 'TT'] = "TT";
    CountryOfResidence[CountryOfResidence["TN"] = 'TN'] = "TN";
    CountryOfResidence[CountryOfResidence["TR"] = 'TR'] = "TR";
    CountryOfResidence[CountryOfResidence["TM"] = 'TM'] = "TM";
    CountryOfResidence[CountryOfResidence["TC"] = 'TC'] = "TC";
    CountryOfResidence[CountryOfResidence["TV"] = 'TV'] = "TV";
    CountryOfResidence[CountryOfResidence["UG"] = 'UG'] = "UG";
    CountryOfResidence[CountryOfResidence["UA"] = 'UA'] = "UA";
    CountryOfResidence[CountryOfResidence["AE"] = 'AE'] = "AE";
    CountryOfResidence[CountryOfResidence["GB"] = 'GB'] = "GB";
    CountryOfResidence[CountryOfResidence["US"] = 'US'] = "US";
    CountryOfResidence[CountryOfResidence["UM"] = 'UM'] = "UM";
    CountryOfResidence[CountryOfResidence["UY"] = 'UY'] = "UY";
    CountryOfResidence[CountryOfResidence["UZ"] = 'UZ'] = "UZ";
    CountryOfResidence[CountryOfResidence["VU"] = 'VU'] = "VU";
    CountryOfResidence[CountryOfResidence["VE"] = 'VE'] = "VE";
    CountryOfResidence[CountryOfResidence["VN"] = 'VN'] = "VN";
    CountryOfResidence[CountryOfResidence["VG"] = 'VG'] = "VG";
    CountryOfResidence[CountryOfResidence["VI"] = 'VI'] = "VI";
    CountryOfResidence[CountryOfResidence["WF"] = 'WF'] = "WF";
    CountryOfResidence[CountryOfResidence["EH"] = 'EH'] = "EH";
    CountryOfResidence[CountryOfResidence["YE"] = 'YE'] = "YE";
    CountryOfResidence[CountryOfResidence["ZM"] = 'ZM'] = "ZM";
    CountryOfResidence[CountryOfResidence["ZW"] = 'ZW'] = "ZW";
    CountryOfResidence[CountryOfResidence["BQ"] = 'BQ'] = "BQ";
    CountryOfResidence[CountryOfResidence["CW"] = 'CW'] = "CW";
    CountryOfResidence[CountryOfResidence["SX"] = 'SX'] = "SX";
    CountryOfResidence[CountryOfResidence["SS"] = 'SS'] = "SS";
    CountryOfResidence[CountryOfResidence["Empty"] = ''] = "Empty";
})(CountryOfResidence = exports.CountryOfResidence || (exports.CountryOfResidence = {}));
