// xero-node/dist/gen/model/payroll-au/leaveCategoryCode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeaveCategoryCode = void 0;
/**
* Code used to identify the Leave Category
*/
var LeaveCategoryCode;
(function (LeaveCategoryCode) {
    LeaveCategoryCode[LeaveCategoryCode["ANNUALLEAVE"] = 'ANNUALLEAVE'] = "ANNUALLEAVE";
    LeaveCategoryCode[LeaveCategoryCode["LONGSERVICELEAVE"] = 'LONGSERVICELEAVE'] = "LONGSERVICELEAVE";
    LeaveCategoryCode[LeaveCategoryCode["PERSONALSICKCARERSLEAVE"] = 'PERSONALSICKCARERSLEAVE'] = "PERSONALSICKCARERSLEAVE";
    LeaveCategoryCode[LeaveCategoryCode["ROSTEREDDAYOFF"] = 'ROSTEREDDAYOFF'] = "ROSTEREDDAYOFF";
    LeaveCategoryCode[LeaveCategoryCode["TIMEOFFINLIEU"] = 'TIMEOFFINLIEU'] = "TIMEOFFINLIEU";
    LeaveCategoryCode[LeaveCategoryCode["COMPASSIONATEANDBEREAVEMENTLEAVE"] = 'COMPASSIONATEANDBEREAVEMENTLEAVE'] = "COMPASSIONATEANDBEREAVEMENTLEAVE";
    LeaveCategoryCode[LeaveCategoryCode["STUDYLEAVE"] = 'STUDYLEAVE'] = "STUDYLEAVE";
    LeaveCategoryCode[LeaveCategoryCode["FAMILYANDDOMESTICVIOLENCELEAVE"] = 'FAMILYANDDOMESTICVIOLENCELEAVE'] = "FAMILYANDDOMESTICVIOLENCELEAVE";
    LeaveCategoryCode[LeaveCategoryCode["SPECIALPAIDLEAVE"] = 'SPECIALPAIDLEAVE'] = "SPECIALPAIDLEAVE";
    LeaveCategoryCode[LeaveCategoryCode["COMMUNITYSERVICELEAVE"] = 'COMMUNITYSERVICELEAVE'] = "COMMUNITYSERVICELEAVE";
    LeaveCategoryCode[LeaveCategoryCode["JURYDUTYLEAVE"] = 'JURYDUTYLEAVE'] = "JURYDUTYLEAVE";
    LeaveCategoryCode[LeaveCategoryCode["DEFENCERESERVELEAVE"] = 'DEFENCERESERVELEAVE'] = "DEFENCERESERVELEAVE";
})(LeaveCategoryCode = exports.LeaveCategoryCode || (exports.LeaveCategoryCode = {}));
