// xero-node/dist/gen/model/files/objectType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectType = void 0;
/**
* The Object Type
*/
var ObjectType;
(function (ObjectType) {
    ObjectType[ObjectType["Unknown"] = 'Unknown'] = "Unknown";
    ObjectType[ObjectType["Accpay"] = 'Accpay'] = "Accpay";
    ObjectType[ObjectType["AccPayCredit"] = 'AccPayCredit'] = "AccPayCredit";
    ObjectType[ObjectType["AccPayPayment"] = 'AccPayPayment'] = "AccPayPayment";
    ObjectType[ObjectType["AccRec"] = 'AccRec'] = "AccRec";
    ObjectType[ObjectType["AccRecCredit"] = 'AccRecCredit'] = "AccRecCredit";
    ObjectType[ObjectType["AccRecPayment"] = 'AccRecPayment'] = "AccRecPayment";
    ObjectType[ObjectType["Adjustment"] = 'Adjustment'] = "Adjustment";
    ObjectType[ObjectType["ApCreditPayment"] = 'ApCreditPayment'] = "ApCreditPayment";
    ObjectType[ObjectType["ApOverPayment"] = 'ApOverPayment'] = "ApOverPayment";
    ObjectType[ObjectType["ApOverPaymentPayment"] = 'ApOverPaymentPayment'] = "ApOverPaymentPayment";
    ObjectType[ObjectType["ApOverPaymentSourcePayment"] = 'ApOverPaymentSourcePayment'] = "ApOverPaymentSourcePayment";
    ObjectType[ObjectType["ApPrepayment"] = 'ApPrepayment'] = "ApPrepayment";
    ObjectType[ObjectType["ApPrepaymentPayment"] = 'ApPrepaymentPayment'] = "ApPrepaymentPayment";
    ObjectType[ObjectType["ApPrepaymentSourcePayment"] = 'ApPrepaymentSourcePayment'] = "ApPrepaymentSourcePayment";
    ObjectType[ObjectType["ArCreditPayment"] = 'ArCreditPayment'] = "ArCreditPayment";
    ObjectType[ObjectType["ArOverPayment"] = 'ArOverPayment'] = "ArOverPayment";
    ObjectType[ObjectType["ArOverpaymentPayment"] = 'ArOverpaymentPayment'] = "ArOverpaymentPayment";
    ObjectType[ObjectType["ArOverpaymentSourcePayment"] = 'ArOverpaymentSourcePayment'] = "ArOverpaymentSourcePayment";
    ObjectType[ObjectType["ArPrepayment"] = 'ArPrepayment'] = "ArPrepayment";
    ObjectType[ObjectType["ArPrepaymentPayment"] = 'ArPrepaymentPayment'] = "ArPrepaymentPayment";
    ObjectType[ObjectType["ArPrepaymentSourcePayment"] = 'ArPrepaymentSourcePayment'] = "ArPrepaymentSourcePayment";
    ObjectType[ObjectType["CashPaid"] = 'CashPaid'] = "CashPaid";
    ObjectType[ObjectType["CashRec"] = 'CashRec'] = "CashRec";
    ObjectType[ObjectType["ExpPayment"] = 'ExpPayment'] = "ExpPayment";
    ObjectType[ObjectType["ManJournal"] = 'ManJournal'] = "ManJournal";
    ObjectType[ObjectType["PurchaseOrder"] = 'PurchaseOrder'] = "PurchaseOrder";
    ObjectType[ObjectType["Receipt"] = 'Receipt'] = "Receipt";
    ObjectType[ObjectType["Transfer"] = 'Transfer'] = "Transfer";
    ObjectType[ObjectType["Account"] = 'Account'] = "Account";
    ObjectType[ObjectType["Contact"] = 'Contact'] = "Contact";
    ObjectType[ObjectType["Business"] = 'Business'] = "Business";
    ObjectType[ObjectType["Employee"] = 'Employee'] = "Employee";
    ObjectType[ObjectType["Person"] = 'Person'] = "Person";
    ObjectType[ObjectType["User"] = 'User'] = "User";
    ObjectType[ObjectType["Org"] = 'Org'] = "Org";
    ObjectType[ObjectType["FixedAsset"] = 'FixedAsset'] = "FixedAsset";
    ObjectType[ObjectType["PayRun"] = 'PayRun'] = "PayRun";
    ObjectType[ObjectType["PriceListItem"] = 'PriceListItem'] = "PriceListItem";
    ObjectType[ObjectType["Bank"] = 'Bank'] = "Bank";
    ObjectType[ObjectType["Current"] = 'Current'] = "Current";
    ObjectType[ObjectType["Equity"] = 'Equity'] = "Equity";
    ObjectType[ObjectType["Expense"] = 'Expense'] = "Expense";
    ObjectType[ObjectType["Fixed"] = 'Fixed'] = "Fixed";
    ObjectType[ObjectType["Liability"] = 'Liability'] = "Liability";
    ObjectType[ObjectType["Prepayment"] = 'Prepayment'] = "Prepayment";
    ObjectType[ObjectType["Revenue"] = 'Revenue'] = "Revenue";
    ObjectType[ObjectType["Sales"] = 'Sales'] = "Sales";
    ObjectType[ObjectType["Overheads"] = 'Overheads'] = "Overheads";
    ObjectType[ObjectType["Depreciatn"] = 'Depreciatn'] = "Depreciatn";
    ObjectType[ObjectType["OtherIncome"] = 'OtherIncome'] = "OtherIncome";
    ObjectType[ObjectType["DirectCosts"] = 'DirectCosts'] = "DirectCosts";
    ObjectType[ObjectType["Currliab"] = 'Currliab'] = "Currliab";
    ObjectType[ObjectType["Termliab"] = 'Termliab'] = "Termliab";
    ObjectType[ObjectType["NonCurrent"] = 'NonCurrent'] = "NonCurrent";
    ObjectType[ObjectType["SalesQuote"] = 'SalesQuote'] = "SalesQuote";
})(ObjectType = exports.ObjectType || (exports.ObjectType = {}));
