// xero-node/dist/gen/model/accounting/taxType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxType = void 0;
/**
* See Tax Types – can only be used on update calls
*/
var TaxType;
(function (TaxType) {
    TaxType[TaxType["OUTPUT"] = 'OUTPUT'] = "OUTPUT";
    TaxType[TaxType["INPUT"] = 'INPUT'] = "INPUT";
    TaxType[TaxType["CAPEXINPUT"] = 'CAPEXINPUT'] = "CAPEXINPUT";
    TaxType[TaxType["EXEMPTEXPORT"] = 'EXEMPTEXPORT'] = "EXEMPTEXPORT";
    TaxType[TaxType["EXEMPTEXPENSES"] = 'EXEMPTEXPENSES'] = "EXEMPTEXPENSES";
    TaxType[TaxType["EXEMPTCAPITAL"] = 'EXEMPTCAPITAL'] = "EXEMPTCAPITAL";
    TaxType[TaxType["EXEMPTOUTPUT"] = 'EXEMPTOUTPUT'] = "EXEMPTOUTPUT";
    TaxType[TaxType["INPUTTAXED"] = 'INPUTTAXED'] = "INPUTTAXED";
    TaxType[TaxType["BASEXCLUDED"] = 'BASEXCLUDED'] = "BASEXCLUDED";
    TaxType[TaxType["GSTONCAPIMPORTS"] = 'GSTONCAPIMPORTS'] = "GSTONCAPIMPORTS";
    TaxType[TaxType["GSTONIMPORTS"] = 'GSTONIMPORTS'] = "GSTONIMPORTS";
    TaxType[TaxType["NONE"] = 'NONE'] = "NONE";
    TaxType[TaxType["INPUT2"] = 'INPUT2'] = "INPUT2";
    TaxType[TaxType["ZERORATED"] = 'ZERORATED'] = "ZERORATED";
    TaxType[TaxType["OUTPUT2"] = 'OUTPUT2'] = "OUTPUT2";
    TaxType[TaxType["CAPEXINPUT2"] = 'CAPEXINPUT2'] = "CAPEXINPUT2";
    TaxType[TaxType["CAPEXOUTPUT"] = 'CAPEXOUTPUT'] = "CAPEXOUTPUT";
    TaxType[TaxType["CAPEXOUTPUT2"] = 'CAPEXOUTPUT2'] = "CAPEXOUTPUT2";
    TaxType[TaxType["CAPEXSRINPUT"] = 'CAPEXSRINPUT'] = "CAPEXSRINPUT";
    TaxType[TaxType["CAPEXSROUTPUT"] = 'CAPEXSROUTPUT'] = "CAPEXSROUTPUT";
    TaxType[TaxType["ECACQUISITIONS"] = 'ECACQUISITIONS'] = "ECACQUISITIONS";
    TaxType[TaxType["ECZRINPUT"] = 'ECZRINPUT'] = "ECZRINPUT";
    TaxType[TaxType["ECZROUTPUT"] = 'ECZROUTPUT'] = "ECZROUTPUT";
    TaxType[TaxType["ECZROUTPUTSERVICES"] = 'ECZROUTPUTSERVICES'] = "ECZROUTPUTSERVICES";
    TaxType[TaxType["EXEMPTINPUT"] = 'EXEMPTINPUT'] = "EXEMPTINPUT";
    TaxType[TaxType["REVERSECHARGES"] = 'REVERSECHARGES'] = "REVERSECHARGES";
    TaxType[TaxType["RRINPUT"] = 'RRINPUT'] = "RRINPUT";
    TaxType[TaxType["RROUTPUT"] = 'RROUTPUT'] = "RROUTPUT";
    TaxType[TaxType["SRINPUT"] = 'SRINPUT'] = "SRINPUT";
    TaxType[TaxType["SROUTPUT"] = 'SROUTPUT'] = "SROUTPUT";
    TaxType[TaxType["ZERORATEDINPUT"] = 'ZERORATEDINPUT'] = "ZERORATEDINPUT";
    TaxType[TaxType["ZERORATEDOUTPUT"] = 'ZERORATEDOUTPUT'] = "ZERORATEDOUTPUT";
    TaxType[TaxType["BLINPUT"] = 'BLINPUT'] = "BLINPUT";
    TaxType[TaxType["DSOUTPUT"] = 'DSOUTPUT'] = "DSOUTPUT";
    TaxType[TaxType["EPINPUT"] = 'EPINPUT'] = "EPINPUT";
    TaxType[TaxType["ES33OUTPUT"] = 'ES33OUTPUT'] = "ES33OUTPUT";
    TaxType[TaxType["ESN33OUTPUT"] = 'ESN33OUTPUT'] = "ESN33OUTPUT";
    TaxType[TaxType["IGDSINPUT2"] = 'IGDSINPUT2'] = "IGDSINPUT2";
    TaxType[TaxType["IMINPUT2"] = 'IMINPUT2'] = "IMINPUT2";
    TaxType[TaxType["MEINPUT"] = 'MEINPUT'] = "MEINPUT";
    TaxType[TaxType["NRINPUT"] = 'NRINPUT'] = "NRINPUT";
    TaxType[TaxType["OPINPUT"] = 'OPINPUT'] = "OPINPUT";
    TaxType[TaxType["OSOUTPUT"] = 'OSOUTPUT'] = "OSOUTPUT";
    TaxType[TaxType["TXESSINPUT"] = 'TXESSINPUT'] = "TXESSINPUT";
    TaxType[TaxType["TXN33INPUT"] = 'TXN33INPUT'] = "TXN33INPUT";
    TaxType[TaxType["TXPETINPUT"] = 'TXPETINPUT'] = "TXPETINPUT";
    TaxType[TaxType["TXREINPUT"] = 'TXREINPUT'] = "TXREINPUT";
    TaxType[TaxType["INPUT3"] = 'INPUT3'] = "INPUT3";
    TaxType[TaxType["INPUT4"] = 'INPUT4'] = "INPUT4";
    TaxType[TaxType["OUTPUT3"] = 'OUTPUT3'] = "OUTPUT3";
    TaxType[TaxType["OUTPUT4"] = 'OUTPUT4'] = "OUTPUT4";
    TaxType[TaxType["SROUTPUT2"] = 'SROUTPUT2'] = "SROUTPUT2";
    TaxType[TaxType["TXCA"] = 'TXCA'] = "TXCA";
    TaxType[TaxType["SRCAS"] = 'SRCAS'] = "SRCAS";
    TaxType[TaxType["BLINPUT2"] = 'BLINPUT2'] = "BLINPUT2";
    TaxType[TaxType["DRCHARGESUPPLY20"] = 'DRCHARGESUPPLY20'] = "DRCHARGESUPPLY20";
    TaxType[TaxType["DRCHARGE20"] = 'DRCHARGE20'] = "DRCHARGE20";
    TaxType[TaxType["DRCHARGESUPPLY5"] = 'DRCHARGESUPPLY5'] = "DRCHARGESUPPLY5";
    TaxType[TaxType["DRCHARGE5"] = 'DRCHARGE5'] = "DRCHARGE5";
    TaxType[TaxType["BADDEBTRELIEF"] = 'BADDEBTRELIEF'] = "BADDEBTRELIEF";
    TaxType[TaxType["IGDSINPUT3"] = 'IGDSINPUT3'] = "IGDSINPUT3";
    TaxType[TaxType["SROVR"] = 'SROVR'] = "SROVR";
    TaxType[TaxType["TOURISTREFUND"] = 'TOURISTREFUND'] = "TOURISTREFUND";
    TaxType[TaxType["TXRCN33INPUT"] = 'TXRCN33INPUT'] = "TXRCN33INPUT";
    TaxType[TaxType["TXRCREINPUT"] = 'TXRCREINPUT'] = "TXRCREINPUT";
    TaxType[TaxType["TXRCESSINPUT"] = 'TXRCESSINPUT'] = "TXRCESSINPUT";
    TaxType[TaxType["TXRCTSINPUT"] = 'TXRCTSINPUT'] = "TXRCTSINPUT";
    TaxType[TaxType["OUTPUTY23"] = 'OUTPUTY23'] = "OUTPUTY23";
    TaxType[TaxType["DSOUTPUTY23"] = 'DSOUTPUTY23'] = "DSOUTPUTY23";
    TaxType[TaxType["INPUTY23"] = 'INPUTY23'] = "INPUTY23";
    TaxType[TaxType["IMINPUT2Y23"] = 'IMINPUT2Y23'] = "IMINPUT2Y23";
    TaxType[TaxType["IGDSINPUT2Y23"] = 'IGDSINPUT2Y23'] = "IGDSINPUT2Y23";
    TaxType[TaxType["TXPETINPUTY23"] = 'TXPETINPUTY23'] = "TXPETINPUTY23";
    TaxType[TaxType["TXESSINPUTY23"] = 'TXESSINPUTY23'] = "TXESSINPUTY23";
    TaxType[TaxType["TXN33INPUTY23"] = 'TXN33INPUTY23'] = "TXN33INPUTY23";
    TaxType[TaxType["TXREINPUTY23"] = 'TXREINPUTY23'] = "TXREINPUTY23";
    TaxType[TaxType["TXCAY23"] = 'TXCAY23'] = "TXCAY23";
    TaxType[TaxType["BADDEBTRELIEFY23"] = 'BADDEBTRELIEFY23'] = "BADDEBTRELIEFY23";
    TaxType[TaxType["IGDSINPUT3Y23"] = 'IGDSINPUT3Y23'] = "IGDSINPUT3Y23";
    TaxType[TaxType["SROVRRSY23"] = 'SROVRRSY23'] = "SROVRRSY23";
    TaxType[TaxType["SROVRLVGY23"] = 'SROVRLVGY23'] = "SROVRLVGY23";
    TaxType[TaxType["SRLVGY23"] = 'SRLVGY23'] = "SRLVGY23";
})(TaxType = exports.TaxType || (exports.TaxType = {}));
