// xero-node/dist/gen/model/accounting/timeZone.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeZone = void 0;
/**
* Timezone specifications
*/
var TimeZone;
(function (TimeZone) {
    TimeZone[TimeZone["AFGHANISTANSTANDARDTIME"] = 'AFGHANISTANSTANDARDTIME'] = "AFGHANISTANSTANDARDTIME";
    TimeZone[TimeZone["ALASKANSTANDARDTIME"] = 'ALASKANSTANDARDTIME'] = "ALASKANSTANDARDTIME";
    TimeZone[TimeZone["ALEUTIANSTANDARDTIME"] = 'ALEUTIANSTANDARDTIME'] = "ALEUTIANSTANDARDTIME";
    TimeZone[TimeZone["ALTAISTANDARDTIME"] = 'ALTAISTANDARDTIME'] = "ALTAISTANDARDTIME";
    TimeZone[TimeZone["ARABIANSTANDARDTIME"] = 'ARABIANSTANDARDTIME'] = "ARABIANSTANDARDTIME";
    TimeZone[TimeZone["ARABICSTANDARDTIME"] = 'ARABICSTANDARDTIME'] = "ARABICSTANDARDTIME";
    TimeZone[TimeZone["ARABSTANDARDTIME"] = 'ARABSTANDARDTIME'] = "ARABSTANDARDTIME";
    TimeZone[TimeZone["ARGENTINASTANDARDTIME"] = 'ARGENTINASTANDARDTIME'] = "ARGENTINASTANDARDTIME";
    TimeZone[TimeZone["ASTRAKHANSTANDARDTIME"] = 'ASTRAKHANSTANDARDTIME'] = "ASTRAKHANSTANDARDTIME";
    TimeZone[TimeZone["ATLANTICSTANDARDTIME"] = 'ATLANTICSTANDARDTIME'] = "ATLANTICSTANDARDTIME";
    TimeZone[TimeZone["AUSCENTRALSTANDARDTIME"] = 'AUSCENTRALSTANDARDTIME'] = "AUSCENTRALSTANDARDTIME";
    TimeZone[TimeZone["AUSCENTRALWSTANDARDTIME"] = 'AUSCENTRALWSTANDARDTIME'] = "AUSCENTRALWSTANDARDTIME";
    TimeZone[TimeZone["AUSEASTERNSTANDARDTIME"] = 'AUSEASTERNSTANDARDTIME'] = "AUSEASTERNSTANDARDTIME";
    TimeZone[TimeZone["AZERBAIJANSTANDARDTIME"] = 'AZERBAIJANSTANDARDTIME'] = "AZERBAIJANSTANDARDTIME";
    TimeZone[TimeZone["AZORESSTANDARDTIME"] = 'AZORESSTANDARDTIME'] = "AZORESSTANDARDTIME";
    TimeZone[TimeZone["BAHIASTANDARDTIME"] = 'BAHIASTANDARDTIME'] = "BAHIASTANDARDTIME";
    TimeZone[TimeZone["BANGLADESHSTANDARDTIME"] = 'BANGLADESHSTANDARDTIME'] = "BANGLADESHSTANDARDTIME";
    TimeZone[TimeZone["BELARUSSTANDARDTIME"] = 'BELARUSSTANDARDTIME'] = "BELARUSSTANDARDTIME";
    TimeZone[TimeZone["BOUGAINVILLESTANDARDTIME"] = 'BOUGAINVILLESTANDARDTIME'] = "BOUGAINVILLESTANDARDTIME";
    TimeZone[TimeZone["CANADACENTRALSTANDARDTIME"] = 'CANADACENTRALSTANDARDTIME'] = "CANADACENTRALSTANDARDTIME";
    TimeZone[TimeZone["CAPEVERDESTANDARDTIME"] = 'CAPEVERDESTANDARDTIME'] = "CAPEVERDESTANDARDTIME";
    TimeZone[TimeZone["CAUCASUSSTANDARDTIME"] = 'CAUCASUSSTANDARDTIME'] = "CAUCASUSSTANDARDTIME";
    TimeZone[TimeZone["CENAUSTRALIASTANDARDTIME"] = 'CENAUSTRALIASTANDARDTIME'] = "CENAUSTRALIASTANDARDTIME";
    TimeZone[TimeZone["CENTRALAMERICASTANDARDTIME"] = 'CENTRALAMERICASTANDARDTIME'] = "CENTRALAMERICASTANDARDTIME";
    TimeZone[TimeZone["CENTRALASIASTANDARDTIME"] = 'CENTRALASIASTANDARDTIME'] = "CENTRALASIASTANDARDTIME";
    TimeZone[TimeZone["CENTRALBRAZILIANSTANDARDTIME"] = 'CENTRALBRAZILIANSTANDARDTIME'] = "CENTRALBRAZILIANSTANDARDTIME";
    TimeZone[TimeZone["CENTRALEUROPEANSTANDARDTIME"] = 'CENTRALEUROPEANSTANDARDTIME'] = "CENTRALEUROPEANSTANDARDTIME";
    TimeZone[TimeZone["CENTRALEUROPESTANDARDTIME"] = 'CENTRALEUROPESTANDARDTIME'] = "CENTRALEUROPESTANDARDTIME";
    TimeZone[TimeZone["CENTRALPACIFICSTANDARDTIME"] = 'CENTRALPACIFICSTANDARDTIME'] = "CENTRALPACIFICSTANDARDTIME";
    TimeZone[TimeZone["CENTRALSTANDARDTIME"] = 'CENTRALSTANDARDTIME'] = "CENTRALSTANDARDTIME";
    TimeZone[TimeZone["CENTRALSTANDARDTIMEMEXICO"] = 'CENTRALSTANDARDTIME(MEXICO)'] = "CENTRALSTANDARDTIMEMEXICO";
    TimeZone[TimeZone["CHATHAMISLANDSSTANDARDTIME"] = 'CHATHAMISLANDSSTANDARDTIME'] = "CHATHAMISLANDSSTANDARDTIME";
    TimeZone[TimeZone["CHINASTANDARDTIME"] = 'CHINASTANDARDTIME'] = "CHINASTANDARDTIME";
    TimeZone[TimeZone["CUBASTANDARDTIME"] = 'CUBASTANDARDTIME'] = "CUBASTANDARDTIME";
    TimeZone[TimeZone["DATELINESTANDARDTIME"] = 'DATELINESTANDARDTIME'] = "DATELINESTANDARDTIME";
    TimeZone[TimeZone["EAFRICASTANDARDTIME"] = 'EAFRICASTANDARDTIME'] = "EAFRICASTANDARDTIME";
    TimeZone[TimeZone["EASTERISLANDSTANDARDTIME"] = 'EASTERISLANDSTANDARDTIME'] = "EASTERISLANDSTANDARDTIME";
    TimeZone[TimeZone["EASTERNSTANDARDTIME"] = 'EASTERNSTANDARDTIME'] = "EASTERNSTANDARDTIME";
    TimeZone[TimeZone["EASTERNSTANDARDTIMEMEXICO"] = 'EASTERNSTANDARDTIME(MEXICO)'] = "EASTERNSTANDARDTIMEMEXICO";
    TimeZone[TimeZone["EAUSTRALIASTANDARDTIME"] = 'EAUSTRALIASTANDARDTIME'] = "EAUSTRALIASTANDARDTIME";
    TimeZone[TimeZone["EEUROPESTANDARDTIME"] = 'EEUROPESTANDARDTIME'] = "EEUROPESTANDARDTIME";
    TimeZone[TimeZone["EGYPTSTANDARDTIME"] = 'EGYPTSTANDARDTIME'] = "EGYPTSTANDARDTIME";
    TimeZone[TimeZone["EKATERINBURGSTANDARDTIME"] = 'EKATERINBURGSTANDARDTIME'] = "EKATERINBURGSTANDARDTIME";
    TimeZone[TimeZone["ESOUTHAMERICASTANDARDTIME"] = 'ESOUTHAMERICASTANDARDTIME'] = "ESOUTHAMERICASTANDARDTIME";
    TimeZone[TimeZone["FIJISTANDARDTIME"] = 'FIJISTANDARDTIME'] = "FIJISTANDARDTIME";
    TimeZone[TimeZone["FLESTANDARDTIME"] = 'FLESTANDARDTIME'] = "FLESTANDARDTIME";
    TimeZone[TimeZone["GEORGIANSTANDARDTIME"] = 'GEORGIANSTANDARDTIME'] = "GEORGIANSTANDARDTIME";
    TimeZone[TimeZone["GMTSTANDARDTIME"] = 'GMTSTANDARDTIME'] = "GMTSTANDARDTIME";
    TimeZone[TimeZone["GREENLANDSTANDARDTIME"] = 'GREENLANDSTANDARDTIME'] = "GREENLANDSTANDARDTIME";
    TimeZone[TimeZone["GREENWICHSTANDARDTIME"] = 'GREENWICHSTANDARDTIME'] = "GREENWICHSTANDARDTIME";
    TimeZone[TimeZone["GTBSTANDARDTIME"] = 'GTBSTANDARDTIME'] = "GTBSTANDARDTIME";
    TimeZone[TimeZone["HAITISTANDARDTIME"] = 'HAITISTANDARDTIME'] = "HAITISTANDARDTIME";
    TimeZone[TimeZone["HAWAIIANSTANDARDTIME"] = 'HAWAIIANSTANDARDTIME'] = "HAWAIIANSTANDARDTIME";
    TimeZone[TimeZone["INDIASTANDARDTIME"] = 'INDIASTANDARDTIME'] = "INDIASTANDARDTIME";
    TimeZone[TimeZone["IRANSTANDARDTIME"] = 'IRANSTANDARDTIME'] = "IRANSTANDARDTIME";
    TimeZone[TimeZone["ISRAELSTANDARDTIME"] = 'ISRAELSTANDARDTIME'] = "ISRAELSTANDARDTIME";
    TimeZone[TimeZone["JORDANSTANDARDTIME"] = 'JORDANSTANDARDTIME'] = "JORDANSTANDARDTIME";
    TimeZone[TimeZone["KALININGRADSTANDARDTIME"] = 'KALININGRADSTANDARDTIME'] = "KALININGRADSTANDARDTIME";
    TimeZone[TimeZone["KAMCHATKASTANDARDTIME"] = 'KAMCHATKASTANDARDTIME'] = "KAMCHATKASTANDARDTIME";
    TimeZone[TimeZone["KOREASTANDARDTIME"] = 'KOREASTANDARDTIME'] = "KOREASTANDARDTIME";
    TimeZone[TimeZone["LIBYASTANDARDTIME"] = 'LIBYASTANDARDTIME'] = "LIBYASTANDARDTIME";
    TimeZone[TimeZone["LINEISLANDSSTANDARDTIME"] = 'LINEISLANDSSTANDARDTIME'] = "LINEISLANDSSTANDARDTIME";
    TimeZone[TimeZone["LORDHOWESTANDARDTIME"] = 'LORDHOWESTANDARDTIME'] = "LORDHOWESTANDARDTIME";
    TimeZone[TimeZone["MAGADANSTANDARDTIME"] = 'MAGADANSTANDARDTIME'] = "MAGADANSTANDARDTIME";
    TimeZone[TimeZone["MAGALLANESSTANDARDTIME"] = 'MAGALLANESSTANDARDTIME'] = "MAGALLANESSTANDARDTIME";
    TimeZone[TimeZone["MARQUESASSTANDARDTIME"] = 'MARQUESASSTANDARDTIME'] = "MARQUESASSTANDARDTIME";
    TimeZone[TimeZone["MAURITIUSSTANDARDTIME"] = 'MAURITIUSSTANDARDTIME'] = "MAURITIUSSTANDARDTIME";
    TimeZone[TimeZone["MIDATLANTICSTANDARDTIME"] = 'MIDATLANTICSTANDARDTIME'] = "MIDATLANTICSTANDARDTIME";
    TimeZone[TimeZone["MIDDLEEASTSTANDARDTIME"] = 'MIDDLEEASTSTANDARDTIME'] = "MIDDLEEASTSTANDARDTIME";
    TimeZone[TimeZone["MONTEVIDEOSTANDARDTIME"] = 'MONTEVIDEOSTANDARDTIME'] = "MONTEVIDEOSTANDARDTIME";
    TimeZone[TimeZone["MOROCCOSTANDARDTIME"] = 'MOROCCOSTANDARDTIME'] = "MOROCCOSTANDARDTIME";
    TimeZone[TimeZone["MOUNTAINSTANDARDTIME"] = 'MOUNTAINSTANDARDTIME'] = "MOUNTAINSTANDARDTIME";
    TimeZone[TimeZone["MOUNTAINSTANDARDTIMEMEXICO"] = 'MOUNTAINSTANDARDTIME(MEXICO)'] = "MOUNTAINSTANDARDTIMEMEXICO";
    TimeZone[TimeZone["MYANMARSTANDARDTIME"] = 'MYANMARSTANDARDTIME'] = "MYANMARSTANDARDTIME";
    TimeZone[TimeZone["NAMIBIASTANDARDTIME"] = 'NAMIBIASTANDARDTIME'] = "NAMIBIASTANDARDTIME";
    TimeZone[TimeZone["NCENTRALASIASTANDARDTIME"] = 'NCENTRALASIASTANDARDTIME'] = "NCENTRALASIASTANDARDTIME";
    TimeZone[TimeZone["NEPALSTANDARDTIME"] = 'NEPALSTANDARDTIME'] = "NEPALSTANDARDTIME";
    TimeZone[TimeZone["NEWFOUNDLANDSTANDARDTIME"] = 'NEWFOUNDLANDSTANDARDTIME'] = "NEWFOUNDLANDSTANDARDTIME";
    TimeZone[TimeZone["NEWZEALANDSTANDARDTIME"] = 'NEWZEALANDSTANDARDTIME'] = "NEWZEALANDSTANDARDTIME";
    TimeZone[TimeZone["NORFOLKSTANDARDTIME"] = 'NORFOLKSTANDARDTIME'] = "NORFOLKSTANDARDTIME";
    TimeZone[TimeZone["NORTHASIAEASTSTANDARDTIME"] = 'NORTHASIAEASTSTANDARDTIME'] = "NORTHASIAEASTSTANDARDTIME";
    TimeZone[TimeZone["NORTHASIASTANDARDTIME"] = 'NORTHASIASTANDARDTIME'] = "NORTHASIASTANDARDTIME";
    TimeZone[TimeZone["NORTHKOREASTANDARDTIME"] = 'NORTHKOREASTANDARDTIME'] = "NORTHKOREASTANDARDTIME";
    TimeZone[TimeZone["OMSKSTANDARDTIME"] = 'OMSKSTANDARDTIME'] = "OMSKSTANDARDTIME";
    TimeZone[TimeZone["PACIFICSASTANDARDTIME"] = 'PACIFICSASTANDARDTIME'] = "PACIFICSASTANDARDTIME";
    TimeZone[TimeZone["PACIFICSTANDARDTIME"] = 'PACIFICSTANDARDTIME'] = "PACIFICSTANDARDTIME";
    TimeZone[TimeZone["PACIFICSTANDARDTIMEMEXICO"] = 'PACIFICSTANDARDTIME(MEXICO)'] = "PACIFICSTANDARDTIMEMEXICO";
    TimeZone[TimeZone["PAKISTANSTANDARDTIME"] = 'PAKISTANSTANDARDTIME'] = "PAKISTANSTANDARDTIME";
    TimeZone[TimeZone["PARAGUAYSTANDARDTIME"] = 'PARAGUAYSTANDARDTIME'] = "PARAGUAYSTANDARDTIME";
    TimeZone[TimeZone["QYZYLORDASTANDARDTIME"] = 'QYZYLORDASTANDARDTIME'] = "QYZYLORDASTANDARDTIME";
    TimeZone[TimeZone["ROMANCESTANDARDTIME"] = 'ROMANCESTANDARDTIME'] = "ROMANCESTANDARDTIME";
    TimeZone[TimeZone["RUSSIANSTANDARDTIME"] = 'RUSSIANSTANDARDTIME'] = "RUSSIANSTANDARDTIME";
    TimeZone[TimeZone["RUSSIATIMEZONE10"] = 'RUSSIATIMEZONE10'] = "RUSSIATIMEZONE10";
    TimeZone[TimeZone["RUSSIATIMEZONE11"] = 'RUSSIATIMEZONE11'] = "RUSSIATIMEZONE11";
    TimeZone[TimeZone["RUSSIATIMEZONE3"] = 'RUSSIATIMEZONE3'] = "RUSSIATIMEZONE3";
    TimeZone[TimeZone["SAEASTERNSTANDARDTIME"] = 'SAEASTERNSTANDARDTIME'] = "SAEASTERNSTANDARDTIME";
    TimeZone[TimeZone["SAINTPIERRESTANDARDTIME"] = 'SAINTPIERRESTANDARDTIME'] = "SAINTPIERRESTANDARDTIME";
    TimeZone[TimeZone["SAKHALINSTANDARDTIME"] = 'SAKHALINSTANDARDTIME'] = "SAKHALINSTANDARDTIME";
    TimeZone[TimeZone["SAMOASTANDARDTIME"] = 'SAMOASTANDARDTIME'] = "SAMOASTANDARDTIME";
    TimeZone[TimeZone["SAOTOMESTANDARDTIME"] = 'SAOTOMESTANDARDTIME'] = "SAOTOMESTANDARDTIME";
    TimeZone[TimeZone["SAPACIFICSTANDARDTIME"] = 'SAPACIFICSTANDARDTIME'] = "SAPACIFICSTANDARDTIME";
    TimeZone[TimeZone["SARATOVSTANDARDTIME"] = 'SARATOVSTANDARDTIME'] = "SARATOVSTANDARDTIME";
    TimeZone[TimeZone["SAWESTERNSTANDARDTIME"] = 'SAWESTERNSTANDARDTIME'] = "SAWESTERNSTANDARDTIME";
    TimeZone[TimeZone["SEASIASTANDARDTIME"] = 'SEASIASTANDARDTIME'] = "SEASIASTANDARDTIME";
    TimeZone[TimeZone["SINGAPORESTANDARDTIME"] = 'SINGAPORESTANDARDTIME'] = "SINGAPORESTANDARDTIME";
    TimeZone[TimeZone["SOUTHAFRICASTANDARDTIME"] = 'SOUTHAFRICASTANDARDTIME'] = "SOUTHAFRICASTANDARDTIME";
    TimeZone[TimeZone["SOUTHSUDANSTANDARDTIME"] = 'SOUTHSUDANSTANDARDTIME'] = "SOUTHSUDANSTANDARDTIME";
    TimeZone[TimeZone["SRILANKASTANDARDTIME"] = 'SRILANKASTANDARDTIME'] = "SRILANKASTANDARDTIME";
    TimeZone[TimeZone["SUDANSTANDARDTIME"] = 'SUDANSTANDARDTIME'] = "SUDANSTANDARDTIME";
    TimeZone[TimeZone["SYRIASTANDARDTIME"] = 'SYRIASTANDARDTIME'] = "SYRIASTANDARDTIME";
    TimeZone[TimeZone["TAIPEISTANDARDTIME"] = 'TAIPEISTANDARDTIME'] = "TAIPEISTANDARDTIME";
    TimeZone[TimeZone["TASMANIASTANDARDTIME"] = 'TASMANIASTANDARDTIME'] = "TASMANIASTANDARDTIME";
    TimeZone[TimeZone["TOCANTINSSTANDARDTIME"] = 'TOCANTINSSTANDARDTIME'] = "TOCANTINSSTANDARDTIME";
    TimeZone[TimeZone["TOKYOSTANDARDTIME"] = 'TOKYOSTANDARDTIME'] = "TOKYOSTANDARDTIME";
    TimeZone[TimeZone["TOMSKSTANDARDTIME"] = 'TOMSKSTANDARDTIME'] = "TOMSKSTANDARDTIME";
    TimeZone[TimeZone["TONGASTANDARDTIME"] = 'TONGASTANDARDTIME'] = "TONGASTANDARDTIME";
    TimeZone[TimeZone["TRANSBAIKALSTANDARDTIME"] = 'TRANSBAIKALSTANDARDTIME'] = "TRANSBAIKALSTANDARDTIME";
    TimeZone[TimeZone["TURKEYSTANDARDTIME"] = 'TURKEYSTANDARDTIME'] = "TURKEYSTANDARDTIME";
    TimeZone[TimeZone["TURKSANDCAICOSSTANDARDTIME"] = 'TURKSANDCAICOSSTANDARDTIME'] = "TURKSANDCAICOSSTANDARDTIME";
    TimeZone[TimeZone["ULAANBAATARSTANDARDTIME"] = 'ULAANBAATARSTANDARDTIME'] = "ULAANBAATARSTANDARDTIME";
    TimeZone[TimeZone["USEASTERNSTANDARDTIME"] = 'USEASTERNSTANDARDTIME'] = "USEASTERNSTANDARDTIME";
    TimeZone[TimeZone["USMOUNTAINSTANDARDTIME"] = 'USMOUNTAINSTANDARDTIME'] = "USMOUNTAINSTANDARDTIME";
    TimeZone[TimeZone["UTC"] = 'UTC'] = "UTC";
    TimeZone[TimeZone["UTC12"] = 'UTC+12'] = "UTC12";
    TimeZone[TimeZone["UTC13"] = 'UTC+13'] = "UTC13";
    TimeZone[TimeZone["UTC02"] = 'UTC02'] = "UTC02";
    TimeZone[TimeZone["UTC08"] = 'UTC08'] = "UTC08";
    TimeZone[TimeZone["UTC09"] = 'UTC09'] = "UTC09";
    TimeZone[TimeZone["UTC11"] = 'UTC11'] = "UTC11";
    TimeZone[TimeZone["VENEZUELASTANDARDTIME"] = 'VENEZUELASTANDARDTIME'] = "VENEZUELASTANDARDTIME";
    TimeZone[TimeZone["VLADIVOSTOKSTANDARDTIME"] = 'VLADIVOSTOKSTANDARDTIME'] = "VLADIVOSTOKSTANDARDTIME";
    TimeZone[TimeZone["VOLGOGRADSTANDARDTIME"] = 'VOLGOGRADSTANDARDTIME'] = "VOLGOGRADSTANDARDTIME";
    TimeZone[TimeZone["WAUSTRALIASTANDARDTIME"] = 'WAUSTRALIASTANDARDTIME'] = "WAUSTRALIASTANDARDTIME";
    TimeZone[TimeZone["WCENTRALAFRICASTANDARDTIME"] = 'WCENTRALAFRICASTANDARDTIME'] = "WCENTRALAFRICASTANDARDTIME";
    TimeZone[TimeZone["WESTASIASTANDARDTIME"] = 'WESTASIASTANDARDTIME'] = "WESTASIASTANDARDTIME";
    TimeZone[TimeZone["WESTBANKSTANDARDTIME"] = 'WESTBANKSTANDARDTIME'] = "WESTBANKSTANDARDTIME";
    TimeZone[TimeZone["WESTPACIFICSTANDARDTIME"] = 'WESTPACIFICSTANDARDTIME'] = "WESTPACIFICSTANDARDTIME";
    TimeZone[TimeZone["WEUROPESTANDARDTIME"] = 'WEUROPESTANDARDTIME'] = "WEUROPESTANDARDTIME";
    TimeZone[TimeZone["WMONGOLIASTANDARDTIME"] = 'WMONGOLIASTANDARDTIME'] = "WMONGOLIASTANDARDTIME";
    TimeZone[TimeZone["YAKUTSKSTANDARDTIME"] = 'YAKUTSKSTANDARDTIME'] = "YAKUTSKSTANDARDTIME";
    TimeZone[TimeZone["YUKONSTANDARDTIME"] = 'YUKONSTANDARDTIME'] = "YUKONSTANDARDTIME";
})(TimeZone = exports.TimeZone || (exports.TimeZone = {}));
